datalab/datalab_session/analysis/source_catalog.py

```diff
@@ -19,23 +36,30 @@ def source_catalog(input: dict):
   x_points = cat_hdu.data["x"][:SOURCE_CATALOG_COUNT]
   y_points = cat_hdu.data["y"][:SOURCE_CATALOG_COUNT]
   flux = cat_hdu.data["flux"][:SOURCE_CATALOG_COUNT]
-  ra = cat_hdu.data["ra"][:SOURCE_CATALOG_COUNT]
-  dec = cat_hdu.data["dec"][:SOURCE_CATALOG_COUNT]
+
+  # ra, dec values may or may not be present in the CAT hdu
+  if "ra" in cat_hdu.data.names and "dec" in cat_hdu.data.names:
+    ra = cat_hdu.data["ra"][:SOURCE_CATALOG_COUNT]
+    dec = cat_hdu.data["dec"][:SOURCE_CATALOG_COUNT]
+  else:
+    # TODO: implement a fallback way to calculate ra, dec from x, y and WCS
+    ra = np.zeros(SOURCE_CATALOG_COUNT)
```

Comment:
I wonder if it would be cleaner to just not return ra/dec if they don't exist, and have the frontend check if they exist to show a message or not?

Reply:
Set them to null and don't include them in the return object, also added a check where they were used to fill out tool tip info in the frontend